src/decomp/decomp_2decompfft.f90

```diff
@@ -64,7 +64,7 @@ subroutine decomposition_2decomp(grid, par)
 
     ! Get local domain size and offset from 2decomp
     grid%cell_dims(:) = xsize(:)
-    par%n_offset(:) = xstart(:)
+    par%n_offset(:) = xstart(:) - 1
```

Comment:
Are you sure about that one? if this is changed then https://github.com/xcompact3d/x3d2/blob/6be7442c2e30ac7e5f15e2ed96d30d1cf6cf9b34/src/mesh.f90#L470 should also be changed I think

Reply:
Was just about writing a comment on this. Yes, I printed the values to make sure and `xstart` from 2decomp returns the global starting index of the array. What we do in coordinates is making sure that (cartesian field(i, j, k) entry @(i=1, j=1, k=1) correspond to the coordinates (0, 0, 0). I think this is the standard in Xcompact3d too. If we use the starting index instead then we would have to substract another 1 to make sure the coordinate corresponding to (i=1, j=1, k=1) is (0, 0, 0). Because TGV is all periodic at the moment we can't really observe a difference and works fine either way, but would be a problem for example in channel case because the initial conditions depend on the global coordinates.